sql/core/src/main/scala/org/apache/spark/sql/execution/python/ExtractPythonUDFs.scala

```diff
@@ -196,8 +195,9 @@ object ExtractPythonUDFs extends Rule[LogicalPlan] with Logging {
   @scala.annotation.tailrec
   private def shouldExtractUDFExpressionTree(e: PythonUDF): Boolean = {
     e.children match {
-      case Seq(child: PythonUDF) => correctEvalType(e) == correctEvalType(child) &&
-        shouldExtractUDFExpressionTree(child)
+      case Seq(child: PythonUDF) =>
+        PythonUDF.correctEvalType(e) == PythonUDF.correctEvalType(child) &&
```

Comment:
Shall we simply import `PythonUDF.correctEvalType` method directly.

Reply:
Sure, done in https://github.com/apache/spark/pull/52238/commits/50664c5e91791f4d9e2c4b65b36be18c9ac746db.